packages/contracts/src/contracts/current/protocol/Exchange/MixinExchangeCore.sol

```diff
@@ -100,29 +96,39 @@ contract MixinExchangeCore is
             expirationTimestampInSec: orderValues[4],
             orderHash: getOrderHash(orderAddresses, orderValues)
         });
+        
+        // Validate order and maker only if first time seen
+        if (filled[order.orderHash] == 0 && cancelled[order.orderHash] == 0) {
+            require(order.makerTokenAmount > 0);
+            require(order.takerTokenAmount > 0);
+            require(isValidSignature(
+                keccak256(orderSchemaHash, order.orderHash),
+                order.maker,
+                signature
+            ));
+        }
+        
+        // Validate taker
+        if (order.taker != address(0)) {
```

Comment:
Is there an efficiency difference between the if block vs `require(order.taker == address(0) || order.taker == msg.sender)`?

Reply:
Not that I know of. It is like this mostly because I reverted the taker abstraction. With taker abstraction we need a more complex expression than `order.taker == msg.sender`.